src/proposer.ts

```diff
@@ -168,6 +200,37 @@ async function getTokenDecimals(tokenAddress: string): Promise<bigint> {
 	}
 }
 
+/**
+ * Fetches token decimals from Sepolia for proper amount calculations.
+ * Returns 18 for ETH (zero address).
+ */
+async function getSepoliaTokenDecimals(tokenAddress: string): Promise<bigint> {
```

Comment:
Let's make sure our handling of token decimals are aligned between OyaChat and the node going forward; will need to closely compare the two.